firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+WebViewDelegates.swift

```diff
@@ -69,21 +69,17 @@ extension BrowserViewController: WKUIDelegate {
         initiatedByFrame frame: WKFrameInfo,
         completionHandler: @escaping () -> Void
     ) {
-        let messageAlert = MessageAlert(message: message, frame: frame)
+        let messageAlert = MessageAlert(message: message, frame: frame, completionHandler: completionHandler)
         if shouldDisplayJSAlertForWebView(webView) {
             logger.log("Javascript message alert will be presented.", level: .info, category: .webview)
 
             present(messageAlert.alertController(), animated: true) {
-                // TODO: [FXIOS-10334] This should be called when the alert is dismissed, not presented
-                completionHandler()
                 self.logger.log("Javascript message alert was completed.", level: .info, category: .webview)
```

Comment:
These logs are in the wrong spot, they'll be logging that the alert was completed even though it was only presented. (So these logs need to be moved also.)

Reply:
Yes true, i'll move those thanks @mattreaganmozilla